prefect_hightouch/syncs/flows.py

```diff
@@ -0,0 +1,199 @@
+    hightouch_credentials: HightouchCredentials,
+    sync_id: str,
+    full_resync: bool = True,
+    max_wait_seconds: int = 900,
+    poll_frequency_seconds: int = 10,
+) -> Sync:
+    """
+    Flow that triggers a sync run and waits for the triggered run to complete.
+
+    Args:
+        hightouch_credentials: Credentials to use for authentication with Hightouch.
+        sync_id: Sync ID used in formatting the endpoint URL.
+        full_resync: Whether to resync all the rows in the query
+            (i.e. ignoring previously synced rows).
+        max_wait_seconds: Maximum number of seconds to wait for the entire
+            flow to complete.
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+
+    Returns:
+        api_models.sync.Sync:
+        - `id: str`
+        - `slug: str`
+        - `workspace_id: str`
+        - `created_at: str`
+        - `updated_at: str`
+        - `destination_id: str`
+        - `model_id: str`
+        - `configuration: Dict`
+        - `schedule: Dict`
+        - `status: "models.SyncStatus"`
+        - `disabled: bool`
+        - `last_run_at: str`
+        - `referenced_columns: List[str]`
+        - `primary_key: str`
+
+    Examples:
+        Trigger a Hightouch sync run and wait for completion as a stand alone flow.
+        ```python
+        import asyncio
+
+        from prefect_hightouch import HightouchCredentials
+        from prefect_hightouch.syncs import trigger_sync_run_and_wait_for_completion
+
+        asyncio.run(
+            trigger_sync_run_and_wait_for_completion(
+                hightouch_credentials=HightouchCredentials(
+                    token="1abc0d23-1234-1a2b-abc3-12ab456c7d8e"
+                ),
+                sync_id=12345,
+                full_resync=True,
+                max_wait_seconds=1800,
+                poll_frequency_seconds=5,
+            )
+        )
+        ```
+
+        Trigger a Hightouch sync run and wait for completion as a subflow.
+        ```python
+        from prefect import flow
+
+        from prefect_hightouch import HightouchCredentials
+        from prefect_hightouch.syncs import trigger_sync_run_and_wait_for_completion
+
+        @flow
+        def sync_flow():
+            hightouch_credentials = HightouchCredentials.load("hightouch-token")
+            sync_metadata = trigger_sync_run_and_wait_for_completion(
+                hightouch_credentials=hightouch_credentials,
+                sync_id=12345,
+                full_resync=True,
+                max_wait_seconds=1800,
+                poll_frequency_seconds=10,
+            )
+            return sync_metadata
+
+        sync_flow()
+        ```
+    """
+    json_body = TriggerRunInput(full_resync=full_resync)
+    sync_run_future = await trigger_run.submit(
+        hightouch_credentials=hightouch_credentials,
+        sync_id=sync_id,
+        json_body=json_body,
+    )
```

Comment:
Can we add a log statement saying that we successfully triggered the sync and potentially give a URL to the Hightouch app to view progress?

Reply:
I don't think I can rebuild it.

Example URL:
```
https://app.hightouch.com/myworkspace-abcdef/sources/1234/runs/123456
```

I have all the IDs, but I am not sure how I can get myworkspace-abcdef

Reply:
I can provide:
`09:59:13.869 | INFO    | Flow run 'fervent-stingray' - Started sync '58784' run '89593592'; open https://app.hightouch.com/ and append /sources/58784/runs/89593592 to view results on webpage.`

Reply:
I made #16 for tracking. We can find a solution to that in a separate PR.